Code/GraphMol/RGroupDecomposition/RGroupCore.cpp

```diff
@@ -54,6 +64,197 @@ void RCore::findIndicesWithRLabel() {
+              case 4:
+                if (!(neighborNumber % 2)) {
+                  switchIt = true;
+                }
+                break;
+              case 3:
+                if (neighborNumber == 1) {
+                  switchIt = true;
+                }
+                break;
+            }
+            // because this neighbor will be moved at the end of the neighbor
+            // list, we need to decrement the neighbor number in case additional
+            // dummy atoms are to be connected to the targetAtom
+            --neighborNumber;
+            if (switchIt) {
+              targetAtom->invertChirality();
+            }
+          }
+        }
+      }
+      for (const int index : hydrogensToAdd) {
+        atomIndicesToKeep.insert(index);
+      }
+#ifdef VERBOSE
+      std::cerr << "Atom Chirality Out " << targetAtom->getChiralTag()
+                << std::endl;
+#endif
+    }
+  }
+  for (const auto newBond : newBonds) {
+    extractedCore->addBond(newBond, true);
+  }
+
+  // Now delete atom's that are not in the core.
+  extractedCore->beginBatchEdit();
+  boost::dynamic_bitset<> removedAtoms(mol.getNumAtoms());
+  for (const auto atom : extractedCore->atoms()) {
+    if (atomIndicesToKeep.find(atom->getIdx()) == atomIndicesToKeep.end()) {
+      extractedCore->removeAtom(atom);
+      removedAtoms.set(atom->getIdx());
+    }
+  }
+  extractedCore->commitBatchEdit();
+
+  // Copy molecule coordinates to extracted core
+  updateSubMolConfs(mol, *extractedCore, removedAtoms);
+
+  for (auto citer = mol.beginConformers(); citer != mol.endConformers();
+       ++citer) {
+    Conformer &newConf =
+        extractedCore->getConformer(static_cast<int>((*citer)->getId()));
+    for (const auto &[fst, snd] : dummyAtomMap) {
+      newConf.setAtomPos(fst->getIdx(), (*citer)->getAtomPos(snd));
+    }
+  }
+
+  // If the molecule has no coordinates and the core does, copy those over
+  if (!mol.getNumConformers() && core->getNumConformers()) {
+    ROMol molCopy(mol);
+    for (auto citer = core->beginConformers(); citer != core->endConformers();
+         ++citer) {
+      const auto newConf = new Conformer(mol.getNumAtoms());
+      newConf->setId((*citer)->getId());
+      newConf->set3D((*citer)->is3D());
+      for (const auto &[fst, snd] : match) {
+        newConf->setAtomPos(snd, (*citer)->getAtomPos(fst));
+      }
+      molCopy.addConformer(newConf);
+    }
+    updateSubMolConfs(molCopy, *extractedCore, removedAtoms);
+    molCopy.clearConformers();
+  }
+
+  extractedCore->clearComputedProps(true);
+  extractedCore->updatePropertyCache(false);
+
+#ifdef VERBOSE
+  std::cerr << "Extracted core smiles " << MolToSmiles(*extractedCore)
+            << std::endl;
+  std::cerr << "Extracted core smarts " << MolToSmarts(*extractedCore)
+            << std::endl;
+#endif
+
+  try {
+    unsigned int failed;
+    MolOps::sanitizeMol(*extractedCore, failed,
```

Comment:
Stopping here for now, this might be causing the issue you referenced with the same structure and different canonicalization.

Reply:
> Stopping here for now, this might be causing the issue you referenced with the same structure and different canonicalization.

Not sure about that- I was trying sanitization here to try and fix the canonicalization issues- but I have made many changes since.